pkg/unittest/valueutils/documentselector.go

```diff
@@ -8,48 +8,75 @@ import (
 )
 
 type DocumentSelector struct {
-	Path  string
-	Value interface{}
+	SkipEmptyTemplates bool `yaml:"skipEmptyTemplates"`
+	MatchMany          bool `yaml:"matchMany"`
+	Path               string
+	Value              interface{}
 }
 
-// FindDocumentIndex, find the index of a document, based on a jsonyamlpath and value
-func (ds DocumentSelector) FindDocumentsIndex(manifests map[string][]common.K8sManifest) (int, error) {
-	indexFound := false
-	actualIndex := -1
-	for _, fileManifests := range manifests {
-		for idx, doc := range fileManifests {
-			var indexError error
-			actualIndex, indexFound, indexError = ds.findDocumentIndex(doc, indexFound, actualIndex, idx)
-			if indexError != nil {
-				return actualIndex, indexError
+func (ds DocumentSelector) SelectDocuments(documentsByTemplate map[string][]common.K8sManifest) (map[string][]common.K8sManifest, error) {
+
+	matchingDocuments := map[string][]common.K8sManifest{}
+	matchingDocumentsCount := 0
+
+	for template, manifests := range documentsByTemplate {
+		filteredManifests, err := ds.selectDocuments(manifests)
+
+		filteredManifestsCount := len(filteredManifests)
+		matchingDocumentsCount += filteredManifestsCount
+
+		if err != nil {
+			return map[string][]common.K8sManifest{}, err
+		}
+
+		if !ds.MatchMany && matchingDocumentsCount > 1 {
+			return map[string][]common.K8sManifest{}, errors.New("multiple indexes found")
+		}
+
+		if filteredManifestsCount > 0 || !ds.SkipEmptyTemplates {
```

Comment:
The idea is that the Selector should decide which documents in which templates should be asserted. 

If the "skipEmptyTemplates" is true - it wont return templates which don't have any matching documents in the resulting map.